internal/crypto/ptls/dialer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package ptls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"time"
+)
+
+type Dialer interface {
+	IsReachableAndTLSValidationSucceeds(
+		address string,
+		certPool *x509.CertPool,
+		logger ErrorOnlyLogger,
+	) error
+}
+
+type ErrorOnlyLogger interface {
```

Comment:
Can this be removed and switch to the `plog.Logger` instead?

Reply:
yes, let's do that